staging/src/kubevirt.io/client-go/api/v1/schema.go

```diff
@@ -463,8 +463,33 @@ type Devices struct {
 	// +optional
 	// +listType=atomic
 	HostDevices []HostDevice `json:"hostDevices,omitempty"`
+	// To configure and access client devices such as redirecting USB
+	// +optional
+	// +listType=atomic
+	ClientDevices []ClientDevice `json:"clientDevices,omitempty"`
+}
+
+// Represent a subset of client devices that can be accessed by VMI. At the
+// moment only, USB devices using Usbredir's library and tooling. Another fit
+// would be a smartcard with libcacard.
+//
+// +k8s:openapi-gen=true
+type ClientDevice struct {
+	// +optional
+	Usb *UsbClientDevice `json:"usb,omitempty"`
 }
 
+// Represents the configuration to USB redirection
+//
+// +k8s:openapi-gen=true
+type UsbClientDevice struct {
+	Name string `json:"name"`
+}
+
+const (
+	UsbClientDeviceMaxNumberOf = 4
```

Comment:
Just curious, why 4 is maximum?

Reply:
That's the supported value in RHV with Spice. I think this can be increased as we could have multiple clients sharing USB with the target VMI but I'd say 4 USB devices is a good start. If we plan to increase in the future, it would be good to evaluate it in QEMU too, if there are other limitations that I'm unaware of.

Reply:
IMO this or some explanation should be in the code as a comment. From glancing at the code it would be hard to figure out why 4 is the maximum value. Changing it would much difficult if we don't know the consequences.